pyproject.toml

```diff
@@ -22,5 +112,87 @@ filterwarnings = [
     # test_config_jupytext_jupyter_fs_meta_manager
     'ignore:The alias `_\(\)` will be deprecated. Use `_i18n\(\)` instead.:FutureWarning',
     # Conda/Python 3.7/Windows - ImportError while loading conftest
-    "ignore:the imp module is deprecated in favour of importlib:DeprecationWarning"
+    "ignore:the imp module is deprecated in favour of importlib:DeprecationWarning",
+    "ignore:Code cell is missing an id field:FutureWarning",
+    "ignore:There is no current event loop:DeprecationWarning",
+]
+
+[tool.coverage.run]
+omit = [
+    "tests/*",
+]
+parallel = true
+branch = false
+
+[tool.coverage.report]
+# Regexes for lines to exclude from consideration
+exclude_lines = [
+    "if self.debug:",
+    # Have to re-enable the standard pragma
+    "pragma: no cover",
+
+    "if __name__ == .__main__.:",
+
+    # Don't complain if tests don't hit defensive assertion code:
+    "raise NotImplementedError",
+    "except ImportError",
+]
+ignore_errors = true
+omit = [
+    "tests/*",
+    "jupytext/version.py",
+    "*/site-packages/*"
+]
+
+# To use tox, see https://tox.readthedocs.io
+# Simply pip or conda install tox
+# If you use conda, you may also want to install tox-conda
+# then run `tox` or `tox -- {pytest args}`
+# To run in parallel using `tox -p` (this does not appear to work for this repo)
+
+# To rebuild the tox environment, for example when dependencies change, use
+# `tox -r`
+
+# Note: if the following error is encountered: `ImportError while loading conftest`
+# then then deleting compiled files has been found to fix it: `find . -name \*.pyc -delete`
+[tool.tox]
```

Comment:
@mwouts What's the plan with `tox`? What's the purpose of it?

My usual reference project uses `nox` and as far as I understand it is used to lint the documentation and examples given there.

Reply:
We have [hatch environments](https://hatch.pypa.io/latest/config/environment/overview/) as an alternative too.

Reply:
Personally I do not test multiple environments locally - I use the GitHub Actions for that. Feel free to remove the references to `tox`. Thanks